lib/_http_client.js

```diff
@@ -282,6 +275,21 @@ function ClientRequest(input, options, cb) {
 util.inherits(ClientRequest, OutgoingMessage);
 
 
+let urlWarningEmitted = false;
+ClientRequest.prototype._emitWarning = function(urlStr) {
+  if (!urlWarningEmitted && !process.noDeprecation) {
+    urlWarningEmitted = true;
+    process.emitWarning(
+      `The provided URL ${urlStr} is not a valid URL, and is supported ` +
+      'in the http module solely for compatibility.',
+      'DeprecationWarning', 'DEP0109');
+  }
+};
+
+ClientRequest.prototype._defaultAgent = function() {
+  return Agent.globalAgent;
+};
```

Comment:
I would avoid exposing those two directly. Maybe we should move them under a private Symbol?